neo/Network/P2P/Payloads/Transaction.cs

```diff
@@ -1,53 +1,38 @@
 using Neo.Cryptography;
 using Neo.IO;
-using Neo.IO.Caching;
 using Neo.IO.Json;
-using Neo.Ledger;
 using Neo.Persistence;
 using Neo.SmartContract;
+using Neo.SmartContract.Native;
 using Neo.VM;
+using Neo.Wallets;
 using System;
 using System.Collections.Generic;
 using System.IO;
 using System.Linq;
-using System.Text;
 
 namespace Neo.Network.P2P.Payloads
 {
-    public abstract class Transaction : IEquatable<Transaction>, IInventory
+    public sealed class Transaction : IEquatable<Transaction>, IInventory
     {
         public const int MaxTransactionSize = 102400;
         /// <summary>
         /// Maximum number of attributes that can be contained within a transaction
         /// </summary>
-        private const int MaxTransactionAttributes = 16;
+        private const int MaxCosigners = 16;
 
-        /// <summary>
-        /// Reflection cache for TransactionType
-        /// </summary>
-        private static ReflectionCache<byte> ReflectionCache = ReflectionCache<byte>.CreateFromEnum<TransactionType>();
-
-        public readonly TransactionType Type;
         public byte Version;
-        public TransactionAttribute[] Attributes;
```

Comment:
We will need Attributes for oracles in the future, we will need to have the hash of downloaded data, for prevent discrepances between the data downloaded by the user and the data downloaded by CN, 
for example:
- Imagine that you made a contract for buy rice, acording to the price of https://rice-price.com , you want to buy it because is at 8$ kilo, but when your TX arrive to CN is 9$, you sign this TX for 8, so this TX should return a FAULT, 
So we should be able to attach the hash of the downloaded data inside the TX

Reply:
The oracles should be implemented in smart contracts.

Reply:
What is your idea for the api rest calls?

Reply:
What do you mean?

Reply:
For the internet access feature, we will need some metadata in blocks or/and transactions, because we will need to store some information for verify.

Reply:
Also, it could be usefull allow to store some information inside the tx, and read this in the smart contract.

Reply:
OK. I will add the `TransactionAttribute` back. But only with two usage: `Script` and `Url`. For other information, I think we should put them in the contract, and calculate fees for them.

Reply:
Sounds good to me

Reply:
Done.